ArcGISRuntimeSDKQt_CppSamples/Routing/RouteAroundBarriers/RouteAroundBarriers.qml

```diff
@@ -0,0 +1,208 @@
+                            text: "Add barriers"
+                            onClicked: {
+                                highlighted = !highlighted;
+                                sampleModel.addBarriers = highlighted;
+
+                                // unset stops button, if set
+                                stopButton.highlighted = false;
+                                sampleModel.addStops = false;
+                            }
+                        }
+
+                    }
+
+                    Column {
+                        spacing: 2
+
+                        CheckBox {
+                            id: bestSequenceBox
+                            text: "Find best sequence"
+                            onCheckedChanged: {
+                                sampleModel.findBestSequence = checked;
+                                sampleModel.createAndDisplayRoute();
+                            }
+                        }
+                        CheckBox {
+                            id: firstStopBox
+                            text: "Preserve first stop"
+                            leftPadding: checkBoxPadding
+                            enabled: bestSequenceBox.checked
+                            onCheckedChanged: {
+                                sampleModel.preserveFirstStop = checked;
+                                sampleModel.createAndDisplayRoute();
+                            }
+                        }
+                        CheckBox {
+                            id: lastStopBox
+                            text: "Preserve last stop"
+                            leftPadding: checkBoxPadding
+                            enabled: bestSequenceBox.checked
+                            onCheckedChanged: {
+                                sampleModel.preserveLastStop = checked;
+                                sampleModel.createAndDisplayRoute();
+                            }
+                        }
+                    }
+
+                    Row {
+                        Layout.alignment: Qt.AlignHCenter
+                        Button {
+                            id: resetButton
+                            text: "Reset"
+                            onClicked: {
+                                sampleModel.clearRouteAndGraphics();
+                                sampleModel.clearDirections();
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Create window for displaying the route directions
+            Rectangle {
+                id: directionWindow
+                Layout.alignment: Qt.AlignBottom
+                Layout.topMargin: 0
+                visible: true
+                Layout.preferredWidth: backBox.width
+                Layout.preferredHeight: 300
+                Layout.margins: 3
+                color: "lightgrey"
+
+                //! [RouteAroundBarriers cpp ListView directionsView]
```

Comment:
> //! [RouteAroundBarriers cpp ListView directionsView]

These are documentation tags. They were in the other sample because the snippet was pulled out into a developer guide doc. Unless there is some doc being written and this is needed from this sample, it should be removed.